plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt

```diff
@@ -105,9 +105,12 @@ class CodeWhispererService {
             return
         }
 
-        if (promptReAuth(project)) return
-
-        if (isQExpired(project)) return
+        if (isQExpired(project)) {
+            ApplicationManager.getApplication().executeOnPooledThread {
+                promptReAuth(project)
```

Comment:
do we not need a return on this?

Reply:
yea, will update

Reply:
as we discussed offline, return only if this reauth attempt fails so we don't fail this current request everytime.

Reply:
updated (and attached screen recording)